compiler/apps/playground/components/Editor/EditorImpl.tsx

```diff
@@ -98,7 +114,46 @@ function parseFunctions(
       suggestions: null,
     });
   }
-  return items;
+
+  return items.map<{
+    compile: boolean;
+    fn: FunctionLike;
+  }>(fn => {
+    const {body} = fn.node;
+    if (t.isBlockStatement(body)) {
+      const selfCheck = checkExplicitMemoizeDirectives(body.directives);
+      if (selfCheck === MemoizeDirectiveState.Enabled)
+        return {compile: true, fn};
+      if (selfCheck === MemoizeDirectiveState.Disabled)
+        return {compile: false, fn};
+
+      const parentWithDirective = fn.findParent(parentPath => {
+        if (parentPath.isBlockStatement() || parentPath.isProgram()) {
+          const directiveCheck = checkExplicitMemoizeDirectives(
+            parentPath.node.directives,
+          );
+          return [
+            MemoizeDirectiveState.Enabled,
+            MemoizeDirectiveState.Disabled,
+          ].includes(directiveCheck);
+        }
+        return false;
+      });
+
+      if (!parentWithDirective) return {compile: true, fn};
+      const parentDirectiveCheck = checkExplicitMemoizeDirectives(
+        (parentWithDirective.node as t.Program | t.BlockStatement).directives,
+      );
+      return {
+        compile: [
+          MemoizeDirectiveState.Enabled,
+          MemoizeDirectiveState.Undefined,
+        ].includes(parentDirectiveCheck),
```

Comment:
nit: This is inefficient since it allocates a new array each time and iterates over it, maybe consider switching `MemoizeDirectiveState` to a module scoped `Set` instead.  `checkExplicitMemoizeDirectives` can just return a string

Reply:
Moved those arrays to separate sets